### DISCUSSION TITLE
Is there a way to hide info and warning messages on marp-cli?

### DISCUSSION THREAD

**Discussion by j2deme** [marp-team/marp#498] [Q&A] [answered]
I'm working on custom Python script to manage my markdown slides, this script includes exporting single slides with Marp, in which case it shows the corresponding info and warning messages, and I don't have any problem with this behavior on this specific section.

My problem is that I also implemented a *batch* exporting, and decided to include a progress bar to show... well the progress, but as the info and warning messages appear they conflict with the progress bar and make it reappear instead of being updated as a single bar.

I hope I have explained myself, anyway, I guess an image is better than thousand words.

![imagen](https://github.com/marp-team/marp/assets/1148641/cf1951a9-3c47-4f69-91a4-96ae46deb494)

If there was a way to hide those messages, I think my interface could behave as expected.

The current line that I'm calling in this section of my script is the following:

```python
os.system(f"marp {SOURCE_DIR}/{slide}.md --output {DIST_DIR}/{slide}.pdf --allow-local-files --pdf-outlines --pdf-outlines.pages=false --pdf-notes")
```

I know there is also an option to user `--input-dir` and `--output`, but if I use that syntax I get even more messages about missing CSS files (which are not missing).

Thanks in advanced.

**Comment by yhatt** [comment, marked as answer]
You can simply suppress the Marp CLI logs that are being output to stderr. Example in the shell:

```bash
marp foobar.md 2>/dev/null
```

I'm not familiar with Python, so providing a clear solution for your use case is difficult for me.

**Reply by j2deme** [reply]
My bad, I forgot to specify that I'm working on Windows.

Now that you mentioned it, it's true, I can use `/dev/null` on Unix based systems, anyway, it was just a matter of finding the alternative for `/dev/null` on Windows and I found that you can use `> NUL 2>&1` at the end of a command line program and it _pipes_ `stderr` and `stdout` to `null`.

I finally got the expected result 😄 

![imagen](https://github.com/marp-team/marp/assets/1148641/57c6922d-5c03-4b9f-b34b-4155cabb25e1)
![imagen](https://github.com/marp-team/marp/assets/1148641/869467ba-ccf4-4604-a2fa-52b517f59efc)
![imagen](https://github.com/marp-team/marp/assets/1148641/e789b7dc-dc4b-4adc-9a65-29248b5f9642)

Thanks for tip!

**Reply by yhatt** [reply]
If you are trying to spawn the CLI programatically, using the language feature to suppress stderr might be a smarter way.

Specifically, you can use the `subprocess` module and set the stderr option to `DEVNULL` to execute the command. It should be a consitent result in different OS.

```python
import subprocess

subprocess.run(["marp foobar.md"], stderr = subprocess.DEVNULL)
```

> I am not experienced in Python, so please note that I cannot guarantee the correctness of this answer.
